immer/detail/arrays/no_capacity.hpp

```diff
@@ -105,6 +106,19 @@ struct no_capacity
         };
     }
 
+    template <typename Iter, typename Sent,
+              std::enable_if_t
+              <is_forward_iterator<Iter> and
+               compatible_sentinel<Iter, Sent>, bool> = true>
+    static no_capacity from_range(Iter first, Sent last)
+    {
+        auto count = static_cast<size_t>(distance(first, last));
```

Comment:
Ahá! So I guess you added the `distance` algorithm such that you can use it here, because the `std` one does not support sentinels. Do your sentinels support `last - first` as a way to compute distance? I'd rather have that than the current proposal.

Reply:
> Do your sentinels support last - first as a way to compute distance?

Good point. Iterator/Sentinel pairs do not necessarily support subtraction, but I can check whether this operation is supported and dispatch accordingly.

Reply:
:+1: